inference-engine/src/inference_engine/ie_core.cpp

```diff
@@ -158,6 +168,46 @@ class Core::Impl : public ICore {
 
     mutable std::map<std::string, InferencePlugin> plugins;
 
+    class CoreConfig final {
+    public:
+        struct CacheConfig {
+            std::shared_ptr<ICacheManager> _cacheManager;
+        };
+        CoreConfig() = default;
+
+        void setAndUpdate(std::map<std::string, std::string>& config) {
+            std::lock_guard<std::mutex> lock(_cacheConfigMutex);
+            auto it = config.find(CONFIG_KEY(CACHE_DIR));
+            if (it != config.end()) {
+                if (!it->second.empty()) {
+                    int err = makedir(it->second.c_str());
+                    if (err != 0 && errno != EEXIST) {
+                        THROW_IE_EXCEPTION << "Couldn't create cache directory ["
+                                           << it->second << "], err=" << strerror(errno) << ")";
+                    }
+                    _cacheConfig._cacheManager = std::make_shared<FileStorageCacheManager>(std::move(it->second));
```

Comment:
may be we can move directory creation to file storage ctor? Or even lazy directory creation once we write to cache first time.

Reply:
E.g. if we call setConfig with CACHE_DIR multiple times, but with different directories, the dummy directories are created, but not used

Reply:
> may be we can move directory creation to file storage ctor? Or even lazy directory creation once we write to cache first time.

In this case, if we have multiple CacheManager's in future - every cache manager will be responsible for folder creation, which looks not necessary

Reply:
@ilyachur how do you think?

Reply:
I agree with Ilya that we directory creation depends on the type of CacheManager, let's move it inside the manager,